samples/apps/logging/logging.cpp

```diff
@@ -843,7 +844,11 @@ namespace loggingapp
         "/log/private/historical",
         HTTP_GET,
         ccf::historical::adapter_v2(
-          get_historical, context.get_historical_state(), is_tx_committed),
+          get_historical,
+          context.get_historical_state(),
+          *context.get_subsystem<ccf::NetworkIdentitySubsystemInterface>()
+             .get(),
+          is_tx_committed),
```

Comment:
This seems like a breaking change in the historical queries API that should be clearly documented in the changelog.

Reply:
The subsystem was Eddy's suggestion and I liked it because it means we don't have to give access to the private key to the application unnecessarily. I chose the `*get` syntax just to make it fit into the same interface pattern as the historical state, which is also a subsystem, but we can e.g. move it into a helper function similar to `get_historical_state`

Reply:
The subsystem is a good idea, but the fact that the historical queries subsystem requires the network identity shouldn't be a worry for the app developers. I think it would be sufficient to pass the network identity to the `ccf::historical::StateCache` the same way that we do for the `network.ledger_secrets` (see `enclave/enclave.h`) so that this is hidden from the app completly.

Reply:
`*ctx.get<>().get()` is unnecessary. Either the callee can take a `shared_ptr` (remove the `*` and `.get()`), or if it needs to take `&`, then we can still remove the final `.get()` (`shared_ptr` implements `operator *` to do the same thing, so we don't need to extract the raw pointer and deref it).

In terms of adding helpers to `get_historical_state`, we should be sure we're only doing that for definitely public subsystems. It's a case of "if the app should use it, here's a helper function to get it". Other possibilities are "it's gettable by the app, but not often enough/definitively enough that we want to give it a helper", and "it's a private subsystem which can _not_ get a publicly visible helper method".

Reply:
With the switch to `adapter_v3` this now takes an `AbstractNodeContext` and it gets the subsystems internally. I think this is the right structure. In future we may want to expose more of the network identity, e.g. the possibility of signing something on behalf of the application, while not providing direct access to the private key, so I believe it's fine to have an abstract interface for this. However, if you don't agree, I'm also happy to remove that helper and make it internal only.

Reply:
Changed my mind again. I kept the subsystem, but made it private for now, but we can easily make it public again by moving the interface header into `include/ccf` if we want to.